### ISSUE TITLE
ldc-1.19.0-beta1 macOS Homebrew build from source fails

### ISSUE THREAD

**Issue by denniscote** [ldc-developers/ldc#3228]
I'm trying to build the ldc-1.19.0-beta1 release using Homebrew on macOS in preparation for revising the Homebrew ldc formula when 1.19.0 is released. The build from source fails because of some missing files. The build process for ldc is beyond me so I am hoping someone can explain what is going wrong. 

The messages I get are:

`dennis@Denniss-iMac homebrew-core % brew install --build-from-source ldc                
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v1.19.0-beta1/ldc-1.19.0-beta1-src.tar.gz
Already downloaded: /Users/dennis/Library/Caches/Homebrew/downloads/0c3960f9ac7ed1d3d64b18ff483c77188571bce4204f1c3dbe09b2f75925cee8--ldc-1.19.0-beta1-src.tar.gz
==> Downloading https://github.com/ldc-developers/ldc/releases/download/v1.19.0-beta1/ldc2-1.19.0-beta1-osx-x86_64.tar.xz
Already downloaded: /Users/dennis/Library/Caches/Homebrew/downloads/a12c1a76eda366598706d38526e85d05c6b1c3309849a81c15de5f80a4dd1496--ldc2-1.19.0-beta1-osx-x86_64.tar.xz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/ldc/1.19.0-beta1 -DCMAKE_BUILD_TYPE=Release -DC
==> make
Last 15 lines from /Users/dennis/Library/Logs/Homebrew/ldc/02.make:
../bin/gen_gccbuiltins: error opening /tmp/ldc-20191116-52559-z828d5/ldc-1.19.0-beta1-src/build/runtime/import/ldc/gccbuiltins_mips.di:No such file or directory
make[2]: *** [runtime/import/ldc/gccbuiltins_arm.di] Error 1
../bin/gen_gccbuiltins: error opening /tmp/ldc-20191116-52559-z828d5/ldc-1.19.0-beta1-src/build/runtime/import/ldc/gccbuiltins_aarch64.di:No such file or directory
make[2]: *** [runtime/import/ldc/gccbuiltins_aarch64.di] Error 1
make[2]: *** [runtime/import/ldc/gccbuiltins_aarch64.di] Error 1
make[1]: *** [runtime/CMakeFiles/druntime-ldc-shared.dir/all] Error 2
make[2]: *** [runtime/import/ldc/gccbuiltins_aarch64.di] Error 1
make[1]: *** [runtime/CMakeFiles/druntime-ldc-debug-shared.dir/all] Error 2
../bin/gen_gccbuiltins: error opening /tmp/ldc-20191116-52559-z828d5/ldc-1.19.0-beta1-src/build/runtime/import/ldc/gccbuiltins_arm.di:No such file or directory
make[2]: *** [runtime/import/ldc/gccbuiltins_mips.di] Error 1
make[2]: *** [runtime/import/ldc/gccbuiltins_aarch64.di] Error 1
make[1]: *** [runtime/CMakeFiles/druntime-ldc-debug.dir/all] Error 2
make[2]: *** [runtime/import/ldc/gccbuiltins_arm.di] Error 1
make[1]: *** [runtime/CMakeFiles/druntime-ldc.dir/all] Error 2
make: *** [all] Error 2
`

I zipped the log files from cmake and make and attached them. The file 02.make has all the messages. 

[logs.zip](https://github.com/ldc-developers/ldc/files/3854449/logs.zip)

Any help or suggestions would be appreciated.

**Comment by kinke** [comment]
I guess this is fixed by:
```diff
diff --git a/runtime/CMakeLists.txt b/runtime/CMakeLists.txt
index 612ab655b..1f7a24552 100644
--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -286,6 +286,8 @@ endif()

 set(GCCBUILTINS "")
 if(TARGET gen_gccbuiltins)
+  file(MAKE_DIRECTORY "${LDC_BUILD_IMPORT_DIR}/ldc")
+
   function(gen_gccbuiltins name)
     set(module "${LDC_BUILD_IMPORT_DIR}/ldc/gccbuiltins_${name}.di")
     if (GCCBUILTINS STREQUAL "")
```

Could you please confirm?

**Comment by denniscote** [comment]
It took some messing around with the Homebrew downloaded files, but yes that does fix it. I have now built 1.19.0-beta1 using the released 1.19.0-beta1 compiler.

```
dennis@Denniss-iMac homebrew-core % ldc2 --version
LDC - the LLVM D compiler (1.19.0):
  based on DMD v2.089.0 and LLVM 9.0.0
  built with LDC - the LLVM D compiler (1.19.0-beta1)
  Default target: x86_64-apple-darwin19.0.0
  Host CPU: skylake
  http://dlang.org - http://wiki.dlang.org/LDC
```


I assume that change will be in the final 1.19.0 source release. 

Thank you for your help, and for all your work on LDC.

**Comment by kinke** [comment]
You're welcome; thx for bothering to test the beta, it clearly does pay off. :) I'll add this.
